### ISSUE TITLE
confit should be able to return bytestrings for Filename()

### ISSUE THREAD

**Issue by jrobeson** [beetbox/beets#2059]

**Comment by sampsyo** [comment]
Can you elaborate a little bit? Where exactly is this needed, and why (vs. just wrapping the lookup with `bytestring_path()`)?

**Comment by jrobeson** [comment]
If you do:

```python
config['foo']['file'] =  some_bytestring   # like we do in beet in various places (likely in the tests)
```
and then attempt
```python
file = config['foo']['file'].as_filename()
```

then it'll blow up right here https://github.com/beetbox/beets/blob/master/beets/util/confit.py#L1350

**Comment by sampsyo** [comment]
Got it. In those cases, it's probably best to change the tests—because, in the real world, those values with be Unicode strings when they come from config files. (And we'd like to test conditions that resemble the real world.)

**Comment by jrobeson** [comment]
ok, so that's back to the age old question of converting bytestrings to utf-8.  come up with a name for the utility function yet? :)

**Comment by sampsyo** [comment]
Right; good point. Let's call it `unicode_path`, I suppose. On Python 3, it can use `surrogateescape`; on Python 2, I suppose it should just use `strict` errors.
